src/components/ComboBox.jsx

```diff
@@ -309,11 +302,8 @@ export default class ComboBox extends Component {
       });
     }
 
-    const domNode = React.findDOMNode(this);
-    domNode.value = value;
-
-    if (this.props.onChange) {
-      this.props.onChange({target: domNode});
+    if (this.props.onUpdate) {
+      this.props.onUpdate({ value: value });
     }
```

Comment:
React components typically do not support onChange callback for valueLink...but I think it should be fine we support onUpdate with valueLink.

With valueLink user can do all value change related handling in _requestChange() method and onUpdate is kind of redundant.